src/apis/EditUserTextEndpoint.js

```diff
@@ -0,0 +1,45 @@
+import apiRequest from '../apiBase/apiRequest';
+import NoModelError from '../apiBase/errors/NoModelError';
+import ValidationError from '../apiBase/errors/ValidationError';
+
+import PostModel from '../models2/PostModel';
+import CommentModel from '../models2/CommentModel';
+
+const TYPE_MAP = {
+  t1: CommentModel,
+  t3: PostModel,
+};
+
+const ENDPOINT = '/api/editusertext';
+
+export default {
+  post(apiOptions, data={}) {
+    const { thingId, text } = data;
+    if (!thingId || !text) {
+      throw new NoModelError(ENDPOINT);
+    }
+
+    const options = {
+      type: 'form',
+      query: {
+        raw_json: 1, // make sure html back from the server is un-escaped
+      },
+      body: {
+        api_type: 'json',
+        text,
+        thing_id: thingId,
+      },
+    };
+
+    return apiRequest(apiOptions, 'POST', ENDPOINT, options)
+      .then(apiResponse => {
+        const { body: { json }} = apiResponse.response;
+        if (json.errors.length) {
+          throw json.errors[0];
```

Comment:
Ah by private I mean `node-private`. What throws me a little about `ValidationError` is the name sounds like a client-side validation issue to me. I think overall its the best approach though, I'll update.